app/src/androidTest/java/org/mozilla/focus/activity/SearchSuggestionTest.java

```diff
@@ -0,0 +1,121 @@
+import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static org.mozilla.focus.utils.AndroidTestUtils.getResourceId;
+
+@Keep
+@RunWith(AndroidJUnit4.class)
+public class SearchSuggestionTest {
+
+    @Rule
+    public final ActivityTestRule<MainActivity> mActivityTestRule = new ActivityTestRule<>(MainActivity.class);
+
+    private Context mContext;
+    private UiDevice mUiDevice;
+
+    @Before
+    public void setUp() throws JSONException {
+        AndroidTestUtils.beforeTest();
+        mContext = InstrumentationRegistry.getInstrumentation().getTargetContext();
+        mUiDevice = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation());
+    }
+
+    @After
+    public void tearDown() {
+        mActivityTestRule.getActivity().finishAndRemoveTask();
+    }
+
+    @Test
+    public void searchSuggestionTest() throws UiObjectNotFoundException, UnsupportedEncodingException {
+
+        SessionLoadedIdlingResource loadingIdlingResource = new SessionLoadedIdlingResource(mActivityTestRule.getActivity());
+
+        // Click search field
+        onView(withId(R.id.home_fragment_fake_input))
+                .perform(click());
+
+        onView(withId(R.id.url_edit)).check(matches(isDisplayed()));
+
+        // Check if the soft keyboard is shown
+        InputMethodManager inputMethodManager = (InputMethodManager) mContext.getSystemService(Context.INPUT_METHOD_SERVICE);
+        Assert.assertTrue(inputMethodManager.isAcceptingText());
+
+        onView(withId(R.id.url_edit)).perform(typeText("rocket"));
+
+        onView(withId(R.id.search_suggestion)).check(matches(isDisplayed()));
+        // Check if the suggestion count is shown at most 5
+        UiCollection suggestionCollection = new UiCollection(new UiSelector().resourceId(getResourceId("search_suggestion")));
+        Assert.assertTrue(suggestionCollection.exists());
+        int suggestionCount = suggestionCollection.getChildCount(new UiSelector().resourceId(getResourceId("suggestion_item")));
+        Assert.assertTrue(suggestionCount <= 5);
+
+        // Pick a suggestion
+        UiObject suggestionItem = suggestionCollection.getChild(new UiSelector().index(new Random().nextInt(suggestionCount)));
+        String suggestionText = suggestionItem.getText();
+        suggestionItem.click();
+
+        // Get default search engine
+        SearchEngine defaultSearchEngine = SearchEngineManager.getInstance().getDefaultSearchEngine(mContext);
+
+        // After page loading completes
+        IdlingRegistry.getInstance().register(loadingIdlingResource);
+
+        UiObject urlBar = mUiDevice.findObject(new UiSelector().resourceId(getResourceId("display_url")));
+        Assert.assertTrue(urlBar.exists());
+
+        // Check if the search result is using default search engine
+        Assert.assertTrue(urlBar.getText().contains(defaultSearchEngine.getIdentifier()));
```

Comment:
Maybe we can find a better way for this check. Cause not every search engine will contain their identifiers in the url. Please add a TODO here and make sure current default search engine is Google in this test case.